### PR TITLE
escrow contract users hook added

### PR DESCRIPTION
**Pull request by sumitnitsurat**
- Added useEscrowContractUsers hook.
- Added handling of network change and account change
- refactored certificate viewer

### CODE REVIEW

File: src/common/hooks/useTitleEscrowUsres.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { useEffect } from "react";
+import { TitleEscrow } from "@govtechsg/token-registry/types/TitleEscrow";
```
Comment (simboonlong) [inline src/common/hooks/useTitleEscrowUsres.ts]: the filename got typo, should be useTitleEscrowUsers.tsx yeah?

File: src/components/ManageAssets/ManageAssets.tsx [hunk 2]
```diff
@@ -1,30 +1,49 @@
...
-    approvedHolderAddress: state.token.approvedHolderAddress,
-    beneficiaryAddress: state.token.beneficiaryAddress,
-    holderAddress: state.token.holderAddress,
-    metamaskAccountError: state.admin.metamaskAccountError,
-  }));
+export const ManageAssets = ({ document, titleEscrow }: ManageAssetsProps) => {
+  const dispatch = useDispatch();
+  const tokenRegistryAddress = getTokenRegistryAddress(document);
+  const { holder, beneficiary } = useTitleEscrowUsers({ titleEscrow });
+
```
Comment (simboonlong) [inline src/components/ManageAssets/ManageAssets.tsx]: what does this means? is it always a must to useTitleEscrowUsers({ titleEscrow }) in order to get holder and beneficiary addresses? then how about approvedBeneficiaryAddress and approvedHolderAddress?

File: src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx [hunk 3]
```diff
@@ -1,17 +1,12 @@
-import React, { useEffect } from "react";
+import React from "react";
 import { TitleView } from "../TitleView";
 import { BLinfo } from "../TitleView/BLinfo";
 import { LoaderSkeleton } from "../../UI/Loader";
 import { TitleEscrow } from "@govtechsg/token-registry/types/TitleEscrow";
-import { useContractFunctionHook } from "@govtechsg/ethers-contract-hook";
+import { useTitleEscrowUsers } from "../../../common/hooks/useTitleEscrowUsres";
```
Comment (simboonlong) [inline src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx]: same, typo

File: src/components/ManageAssets/ManageAssets.tsx [hunk 4]
```diff
@@ -44,8 +63,8 @@ export const ManageAssets = ({ document }: ManageAssetsProps) => {
                 userWalletAddress={userWalletAddress}
                 approvedBeneficiaryAddress={approvedBeneficiaryAddress}
                 approvedHolderAddress={approvedHolderAddress}
-                beneficiaryAddress={beneficiaryAddress}
-                holderAddress={holderAddress}
+                beneficiaryAddress={beneficiary || ""}
+                holderAddress={holder || ""}
```
Comment (simboonlong) [inline src/components/ManageAssets/ManageAssets.tsx]: um think related to the qn above, why approvedBeneficiaryAddress and approvedHolderAddress don't have to put the empty string fallback? as in what is the difference in fetching approach between these 2 sets of addresses?

File: src/common/hooks/useTitleEscrowUsres.ts (hunk 1 above)
Comment (sumitnitsurat) [reply src/common/hooks/useTitleEscrowUsres.ts]: fixed it

File: src/components/ManageAssets/ManageAssets.tsx (hunk 2 above)
Comment (sumitnitsurat) [reply src/components/ManageAssets/ManageAssets.tsx]: yupp..pass the instance of approved escrow contract to get the approved holder and beneficiary.

File: src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx (hunk 3 above)
Comment (sumitnitsurat) [reply src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx]: fixed

File: src/components/ManageAssets/ManageAssets.tsx (hunk 4 above)
Comment (sumitnitsurat) [reply src/components/ManageAssets/ManageAssets.tsx]: added empty fallback.

**Pull request closed without merging**